table/tables/tables.go

```diff
@@ -323,6 +323,20 @@ func (t *Table) rebuildIndices(ctx context.Context, h int64, touched []bool, old
 	return nil
 }
 
+func (t *Table) delDirtyIndexKey(createdIdx int, recordID int64, r []interface{}, txn kv.Transaction) {
+	for i, v := range t.indices {
+		if v == nil {
+			continue
+		}
+
+		if i >= createdIdx {
+			break
+		}
+		colVals, _ := v.FetchValues(r)
+		_ = v.X.Delete(txn, colVals, recordID)
```

Comment:
Yes you are right. It is ok here. 
But when executing update statements, you will meet similar problem. There is also a bug in update statement, we update row data before updating index. So if meet keyexist error when updating index, unionstore is still dirty.
So I think it may be better to provide some mechanism to check index constraints before operating unionstore.